src/traits.rs

```diff
@@ -0,0 +1,24 @@
+use std::os::unix::io::RawFd;
+use std::convert::TryFrom;
+
+use daemon::*;
+
+impl TryFrom<RawFd> for SocketType {
```

Comment:
We can put this directly into `daemon.rs`.

Reply:
I had issues placing this impl in `daemon.rs`, the `type Error = &'static str;` conflicted with the crate Error type. Any suggestions?

Reply:
@jabedude I directly pushed a commit on top of your repository and branch, moving the trait to the other file (plus rustfmt). Feel free to squash everything together.